packages/server/graphql/private/mutations/generateMeetingSummary.ts

```diff
@@ -0,0 +1,185 @@
+import yaml from 'js-yaml'
+import getRethink from '../../../database/rethinkDriver'
+import MeetingRetrospective from '../../../database/types/MeetingRetrospective'
+import getKysely from '../../../postgres/getKysely'
+import OpenAIServerManager from '../../../utils/OpenAIServerManager'
+import getPhase from '../../../utils/getPhase'
+import {MutationResolvers} from '../resolverTypes'
+
+const generateMeetingSummary: MutationResolvers['generateMeetingSummary'] = async (
+  _source,
+  {teamIds, prompt},
+  {dataLoader}
+) => {
+  const r = await getRethink()
+  const pg = getKysely()
+  const MIN_MILLISECONDS = 60 * 1000 // 1 minute
+  const MIN_REFLECTION_COUNT = 3
+
+  const endDate = new Date()
+  const twoYearsAgo = new Date()
+  twoYearsAgo.setFullYear(endDate.getFullYear() - 2)
+
+  const rawMeetings = (await r
+    .table('NewMeeting')
+    .getAll(r.args(teamIds), {index: 'teamId'})
+    .filter((row: any) =>
+      row('meetingType')
+        .eq('retrospective')
+        .and(row('createdAt').ge(twoYearsAgo))
```

Comment:
I have this two year limit so that the mutation doesn't take too long, but we could remove it and just accept that it might take several minutes to complete? Otherwise, when building the Insights UI, we'll need to restrict summaries to the past two years